resources/datahub/converter.go

```diff
@@ -57,7 +58,47 @@ func fromModelToRequest(model awsDatahubResourceModel, ctx context.Context) *dat
 		igs = append(igs, ig)
 	}
 	req.InstanceGroups = igs
-	debug(ctx, "Conversion from awsDatahubResourceModel to CreateAWSClusterRequest has finished.")
+	debug(ctx, "Conversion from datahubResourceModel to CreateAWSClusterRequest has finished.")
+	return &req
+}
+
+func fromModelToAzureRequest(model datahubResourceModel, ctx context.Context) *datahubmodels.CreateAzureClusterRequest {
+	debug(ctx, "Conversion from datahubResourceModel to CreateAzureClusterRequest started.")
+	req := datahubmodels.CreateAzureClusterRequest{}
+	req.ClusterName = model.Name.ValueString()
+	req.ClusterTemplateName = model.ClusterTemplate.ValueString()
+	req.EnvironmentName = model.Environment.ValueString()
+	req.ClusterDefinitionName = model.ClusterDefinition.ValueString()
+	var igs []*datahubmodels.AzureInstanceGroupRequest
+	debug(ctx, fmt.Sprintf("%d instance group found in the input model.", len(model.InstanceGroup)))
+	for _, group := range model.InstanceGroup {
+		debug(ctx, fmt.Sprintf("Converting InstanceGroupRequest: %+v.", group))
+		var volReqs []*datahubmodels.AttachedVolumeRequest
+		debug(ctx, fmt.Sprintf("%d attached volume request found in the input model.", len(model.InstanceGroup)))
+		for _, vrs := range group.AttachedVolumeConfiguration {
+			debug(ctx, fmt.Sprintf("Converting AttachedVolumeConfiguration: %+v.", vrs))
+			volReqs = append(volReqs, createAttachedVolumeRequest(vrs))
+		}
+		var igRecipes []string
+		if group.Recipes != nil && len(group.Recipes) > 0 {
+			for _, recipe := range group.Recipes {
+				igRecipes = append(igRecipes, recipe.ValueString())
+			}
+		}
+		rootVolumeSize := int32(group.RootVolumeSize.ValueInt64())
```

Comment:
One is `int32` and the other one is `int64`. Why are we not simply declaring them to be the same width?

Reply:
the `types` package doesn't contain an Int32 type, just the Int64